tools/generate-templates.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -euo pipefail
+
+rm -rf test/roles/* || true
+mkdir -p test/roles
```

Comment:
Every time we run `tox -e lint`, it will generate some temporary files locally, can we add `test/roles` in `.gitignore`?

Reply:
pre-commit will ignore stuff inside .gitignore

Reply:
I may be missing something but why not creating a temporary directory with mktemp ? issue with non linux os like macosx ?

Reply:
What benefits`mktemp` can bring?

IMHO, `mktemp` is better for `unique` file name. In this PR, we just want to create a directory and generate some files in them.

Reply:
I find it cleaner. It won't do any change in the working directory and it should avoid having to deal with configurations files of tools like git or pre-commit to ignore it.

Reply:
hm. maybe it would be easier to fix all errors before merging this PR, _maybe_ this would make the .ansible-lint-ignore file useless ?

Reply:
Some errors are unfixable. And the fixable ones are too much to be included in this PR. there's no going around `.ansible-lint-ignore` file

Reply:
It's fun to fix the real lint issue in another issue.

Ignoring them to make the PR smaller is acceptable.